community/modules/compute/gke-node-pool/threads_per_core_calc.tf

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# This file is meant to be reused by multiple modules.
+# "description": Allows for 'threads_per_core=0: SMT will be disabled where compatible (default)'
+
+# "inputs":
+# var.machine_type: Machine type for the instance being evaluated.
+# var.threads_per_core : Sets the number of threads per physical core, where 0
+#                        has behavior described in description.
+
+# "outputs":
+# local.set_threads_per_core: bool that tells if threads per core should be set,
+#                             to be used with a dynamic block.
+# local.threads_per_core: actual threads_per_core to be used.
+
+locals {
+  machine_vals            = split("-", var.machine_type)
+  machine_family          = local.machine_vals[0]
+  machine_not_shared_core = length(local.machine_vals) > 2
```

Comment:
I suppose you are trying to eliminate `e2-small` and such. Wondering if there is a more explicit manner.